packages/webapp/src/util/convert-units/applicationRates.ts

```diff
@@ -0,0 +1,223 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Measure } from 'convert-units';
+
+export type ApplicationRateVolumeUnits =
+  | 'l/ha'
+  | 'ml/ha'
+  | 'l/m2'
+  | 'ml/m2'
+  | 'gal/ac'
+  | 'gal/ft2'
+  | 'fl-oz/ft2'
+  | 'fl-oz/ac';
+
+export type ApplicationRateWeightUnits =
+  | 'kg/ha'
+  | 'g/ha'
+  | 'mt/ha'
+  | 'g/m2'
+  | 'kg/m2'
+  | 'mt/m2'
+  | 'lb/ac'
+  | 'lb/ft2'
+  | 'oz/ft2'
+  | 't/ft2'
+  | 'oz/ac'
+  | 't/ac';
+
+// Custom measures; https://github.com/convert-units/convert-units?tab=readme-ov-file#custom-measures
+export const applicationRateVolume: Measure<'metric' | 'imperial', ApplicationRateVolumeUnits> = {
+  systems: {
+    metric: {
+      'l/ha': {
+        name: {
+          singular: 'Liter per Hectare',
+          plural: 'Liters per Hectare',
+        },
+        to_anchor: 1,
+      },
+      'ml/ha': {
+        name: {
+          singular: 'Milliliter per Hectare',
+          plural: 'Milliliters per Hectare',
+        },
+        to_anchor: 0.001,
+      },
+      'l/m2': {
+        name: {
+          singular: 'Liter per Square Meter',
+          plural: 'Liters per Square Meter',
+        },
+        to_anchor: 1 * 10000, // volume to volume conversion x reciprocal of area-to-area conversion
+      },
+      'ml/m2': {
+        name: {
+          singular: 'Milliliter per Square Meter',
+          plural: 'Milliliters per Square Meter',
+        },
+        to_anchor: 0.001 * 10000,
+      },
+    },
+    imperial: {
+      'gal/ac': {
+        name: {
+          singular: 'Gallon per Acre',
+          plural: 'Gallons per Acre',
+        },
+        to_anchor: 1,
+      },
+      'gal/ft2': {
+        name: {
+          singular: 'Gallon per Square Foot',
+          plural: 'Gallons per Square Foot',
+        },
+        to_anchor: 1 * 43560, // volume-to-volume conversion x reciprocal of area-to-area conversion
+      },
+      'fl-oz/ft2': {
+        name: {
+          singular: 'Fluid Ounce per Square Foot',
+          plural: 'Fluid Ounces per Square Foot',
+        },
+        to_anchor: (1 / 128) * 43560,
+      },
+      'fl-oz/ac': {
+        name: {
+          singular: 'Fluid Ounce per Acre',
+          plural: 'Fluid Ounces per Acre',
+        },
+        to_anchor: 1 / 128,
+      },
+    },
+  },
+  anchors: {
+    metric: {
+      imperial: {
+        ratio: 0.264172 / 2.47105, // l/gal * (ac/ha)
```

Comment:
Above you used (1/ 128) I think to be clear. You can consider doing the same here.

Reply:
1/128 is the ratio of fluid ounces to gallons, but the numbers here are the ratios of litres to gallons + acres to hectares and there are no clean proportions for those (only approximations) as they are entirely different systems.

Reply:
I am saying that this would be 0.264172 *( 1 / 2.47105), since the below line is 0.264172 * 2.47105. But if thats not what was happening pls ignore!

Reply:
Ah I see, I see 🤔 Yes I wasn't consistent with that.

I think I like how it's done in the weight anchors (system conversion) below, then. And I'll use the more intuitive litre to gallon ratio so that would be

`( 1 /  3.78541 ) * 2.47105`

on this line then

Gosh it's stressing me out how the denominator isn't flipping though, I need to check that, although calculation wise this is what worked out 😅

Reply:
Ok, done! Made both changes -- kept the common constant between the imperial to metric and metric to imperial, and used ratio multiplication instead of division. Can you check it @Duncan-Brain?

Reply:
Whatever is clear to you, I am not exactly sure why it stood out to me. 

Thinking about it more than I did the first time, I think for precision less math operations is better. For the factor itself I think what I saw was that it was showcasing a well known conversion factor. So I might expect: (wellKnownImperialToMetricConversionFactorForX) * (1/wellKnownImperialToMetricConversionFactorForY) OR 
(wellKnownMetricToImperialConversionFactorForX) * (1/wellKnownMetricToImperialConversionFactorForY)

If that exists or make sense but its good as is!